### DISCUSSION TITLE
VPN client on Mac with EAP-MSCHAPv2

### DISCUSSION THREAD

**Discussion by qubu** [strongswan/strongswan#1059] [Q&A] [answered]
Hi,

I have a question regarding support of EAP-MSCHAPv2 authentication on macOS Monterey. Unfortunately, manually created connection from System Preferences does not work. Configuration of command-line strongswan client (which is working) is pretty simple `ipsec.conf`:
``` 
conn xxxx
        leftfirewall=yes
	leftauth=eap-mschapv2
	eap_identity=xxxx
	right=yyyyyy.xxxxx.com
	rightauth=pubkey
	rightid=%any
	rightsubnet=X.X.X.X/24
	auto=start
	dpdaction=restart
	closeaction=restart
	keyingtries=%forever
	leftsourceip=%config
```

How can I mapped it into native Mac VPN client? certificates are imported to Mac's Keychain.

Thanks
Kuba

**Comment by tobiasbrunner** [comment]
> Unfortunately, manually created connection from System Preferences does not work

Why not?

> How can I mapped it into native Mac VPN client? 

Should be straight forward.

**Reply by qubu** [reply]
In macOS log I saw "authentication failed" message in response to IKE_AUTH.

**Comment by qubu** [comment, marked as answer]
Meantime, I've found the solution. I have simply skipped RemoteID field in Mac VPN Client GUI in Network Preferences. Unfortunately, to do so, you must generate a mobile profile.

**Reply by tobiasbrunner** [reply]
Not sure I get what you are saying. Because at least on iOS the remote identity can be configured directly in the GUI.

**Reply by qubu** [reply]
Yes, but in may case for `rightid=%any` I have to leave Remote ID empty. GUI does not allow it.
<img width="350" alt="Screenshot 2022-05-24 at 17 34 16" src="https://user-images.githubusercontent.com/1298991/170075234-fc357add-99cf-4265-80d0-23065e7db5a9.png">

**Reply by tobiasbrunner** [reply]
Why would you need to leave it empty? Just configure the actual identity (it's safer to do so anyway).
